gdal/ogr/ogrsf_frmts/gpkg/ogrgeopackagetablelayer.cpp

```diff
@@ -4431,14 +4394,14 @@ OGRErr OGRGeoPackageTableLayer::RecreateTable(const CPLString& osColumnsForCreat
         "SELECT sql FROM sqlite_master WHERE type IN ('trigger','index') "
         "AND lower(tbl_name)=lower('%q') LIMIT 10000",
         m_pszTableName );
-    SQLResult oTriggers;
-    OGRErr eErr = SQLQuery(hDB, pszSQL, &oTriggers);
+    OGRErr eErr = OGRERR_NONE;
+    auto oTriggers = SQLQuery(hDB, pszSQL);
     sqlite3_free(pszSQL);
 
 /* -------------------------------------------------------------------- */
 /*      Make a temporary table with new content.                        */
 /* -------------------------------------------------------------------- */
-    if( eErr == OGRERR_NONE )
+    if( oTriggers )
```

Comment:
eErr shoud be set to OGRERR_FAILURE when oTriggers == null

Reply:
Good catch, fixed it